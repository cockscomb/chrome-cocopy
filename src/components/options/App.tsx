import {h} from 'preact';
import {ThemeProvider} from 'styled-components';

import {Switch, Route} from 'react-router-dom';

import {theme} from '../common/Theme';
import {MainColumn, Title, VersionLabel} from './Parts';
import {FunctionList} from './FunctionList';
import {Links} from './Links';
import {Hint, DebuggingHint} from './Hints';
import {InstallFunction} from './InstallFunction';

export const App = () => {
  return (
    <ThemeProvider theme={theme}>
      <MainColumn>
        <Title />
        <Switch>
          <Route exact path="/">
            <FunctionList />
            <Hint />
            <DebuggingHint />
            <Links />
          </Route>
          <Route path="/install">
            <InstallFunction />
            <Hint />
            <Links />
          </Route>
        </Switch>
        <VersionLabel />
      </MainColumn>
    </ThemeProvider>
  );
};
