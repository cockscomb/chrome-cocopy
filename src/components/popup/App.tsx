import {h} from 'preact';
import {ThemeProvider} from 'styled-components';

import {theme} from '../common/Theme';
import {PopupHeader} from './Header';
import {FunctionList} from './FunctionList';

export const App = () => {
  return (
    <ThemeProvider theme={theme}>
      <div>
        <PopupHeader />
        <FunctionList />
      </div>
    </ThemeProvider>
  );
};
