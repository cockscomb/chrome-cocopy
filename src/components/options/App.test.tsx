import {h} from 'preact';
import render from 'preact-render-to-string';

import {App} from './App';

test('render options', () => {
  expect(render(<App />, {}, {pretty: true})).toMatchSnapshot();
});
