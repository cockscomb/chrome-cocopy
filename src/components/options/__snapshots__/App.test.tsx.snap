// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render options 1`] = `
.c0 {
  padding: 0;
  width: 840px;
  margin: 20px auto;
}

.c1 {
  display: inline-block;
  background: no-repeat center/contain url('img/logo.png');
  width: 123px;
  height: 60px;
}

.c2 {
  margin-bottom: 2rem;
}

.c3 {
  font-family: Menlo,Monaco,Consolas,"Liberation Mono","Courier New",monospace;
  font-size: 1.25rem;
}

.c4 {
  margin-left: 0.5rem;
}

.c15 {
  font-size: .875rem;
  margin-bottom: 1rem;
}

.c15 li {
  margin-bottom: 0.5rem;
}

.c15 code {
  font-size: .75rem;
  font-family: Menlo,Monaco,Consolas,"Liberation Mono","Courier New",monospace;
  background-color: #f5f2f0;
}

.c16 {
  text-align: center;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 380px;
  height: 32px;
  padding: 0.25rem 0.5rem;
  color: #010203;
  background-color: #FFF;
  cursor: pointer;
}

.c13:focus,
.c13:hover {
  -webkit-filter: brightness(110%);
  filter: brightness(110%);
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 380px;
  height: 32px;
  padding: 0.25rem 0.5rem;
  color: #000000;
  background-color: #f5f5f5;
  cursor: pointer;
}

.c7:focus,
.c7:hover {
  -webkit-filter: brightness(110%);
  filter: brightness(110%);
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 380px;
  height: 32px;
  padding: 0.25rem 0.5rem;
  color: #FFFFFF;
  background-color: #ff5722;
  cursor: pointer;
}

.c11:focus,
.c11:hover {
  -webkit-filter: brightness(110%);
  filter: brightness(110%);
}

.c8 {
  width: 32px;
}

.c9 {
  font-size: 1rem;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c14 {
  border: 1px solid #010203;
}

.c14:hover {
  color: #007BFF;
  border-color: #007BFF;
}

.c5 {
  opacity: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 2.25rem;
  height: 32px;
  cursor: pointer;
}

.c10 {
  width: 2.25rem;
  opacity: 1;
  cursor: move;
}

.c12 {
  margin-top: 0.5rem;
}

<body>
  <div>
    <div
      class="c0"
    >
      <header>
        <a
          href="#/"
        >
          <div
            class="c1"
          />
        </a>
      </header>
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          Functions
        </h2>
        <div
          class="c4"
        >
          <div>
            <div
              class="c5"
            >
              <div
                class="c6"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-caret-right fa-w-6 fa-lg "
                  data-icon="caret-right"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 192 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
                    fill="currentColor"
                  />
                </svg>
              </div>
              <div
                class="c7"
                tabindex="-1"
              >
                <div
                  class="c8"
                />
                <div
                  class="c9"
                >
                  Markdown: [title](url)
                </div>
              </div>
              <div
                draggable="true"
              >
                <div
                  class="c6 c10"
                  draggable="true"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-bars fa-w-14 "
                    data-icon="bars"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              class="c5"
            >
              <div
                class="c6"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-caret-right fa-w-6 fa-lg "
                  data-icon="caret-right"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 192 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
                    fill="currentColor"
                  />
                </svg>
              </div>
              <div
                class="c11"
                tabindex="-1"
              >
                <div
                  class="c8"
                />
                <div
                  class="c9"
                >
                  HTML: &lt;a href={url}&gt;{title}&lt;/a&gt;
                </div>
              </div>
              <div
                draggable="true"
              >
                <div
                  class="c6 c10"
                  draggable="true"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-bars fa-w-14 "
                    data-icon="bars"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
          <div
            class="c12"
          >
            <div
              class="c5"
            >
              <div
                class="c6"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-plus fa-w-14 "
                  data-icon="plus"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                    fill="currentColor"
                  />
                </svg>
              </div>
              <div
                class="c13 c14"
              >
                <div
                  class="c8"
                />
                <div
                  class="c9"
                >
                  Create New Function
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          Hints
        </h2>
        <div
          class="c4"
        >
          <ul
            class="c15"
          >
            <li>
              The code must be a single function that returns a string value to copy.
            </li>
            <li>
              <code>
                render(template, view)
              </code>
               - You can render
               
              <a
                href="https://github.com/janl/mustache.js"
                rel="noreferrer noopener"
                target="_blank"
              >
                mustache
              </a>
               
              templates.
            </li>
            <li>
              <code>
                new DOMParser().parseFromString(content, 'text/html')
              </code>
              - You can use
               
              <a
                href="https://developer.mozilla.org/docs/Web/API/DOMParser"
                rel="noreferrer noopener"
                target="_blank"
              >
                <code>
                  DOMParser
                </code>
              </a>
               
              to query the page document.
            </li>
            <li>
              To feedback an error, use 
              <code>
                throw new Error(...)
              </code>
              . It will appear at the function that produced the error.
            </li>
            <li>
              The code runs safely under the
               
              <a
                href="https://developer.chrome.com/apps/sandboxingEval"
                rel="noreferrer noopener"
                target="_blank"
              >
                sandbox
              </a>
              , not under the page.
            </li>
            <li>
              Visit
               
              <a
                href="https://github.com/pokutuna/chrome-cocopy/blob/master/gallery/README.md"
                rel="noreferrer noopener"
                target="_blank"
              >
                Function Gallery
              </a>
               
              to find new function and see sample codes.
            </li>
          </ul>
        </div>
      </div>
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          Debugging
        </h2>
        <div
          class="c4"
        >
          <ul
            class="c15"
          >
            <li>
              To open the developer console by right clicking on the popup & selecting "Inspect"
            </li>
            <li>
              You can find the code on the "Sources" panel, in
               
              <code>
                sandbox.html
              </code>
               &gt; 
              <code>
                (no domain)
              </code>
               after executing it.
            </li>
            <li>
              You also can use 
              <code>
                debugger;
              </code>
               in the code.
            </li>
          </ul>
        </div>
      </div>
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          Links
        </h2>
        <div
          class="c4"
        >
          <ul
            class="c15"
          >
            <li>
              <a
                href="https://chrome.google.com/webstore/detail/cocopy/ihnfodlbkhgjnbheemjhkjfkfglgbdgc"
                rel="noreferrer noopener"
                target="_blank"
              >
                Chrome Web Store
              </a>
            </li>
            <li>
              <a
                href="https://github.com/pokutuna/chrome-cocopy"
                rel="noreferrer noopener"
                target="_blank"
              >
                GitHub Repository
              </a>
            </li>
            <li>
              <a
                href="https://github.com/pokutuna/chrome-cocopy/blob/master/gallery/README.md"
                rel="noreferrer noopener"
                target="_blank"
              >
                Function Gallery
              </a>
            </li>
            <li>
              <a
                href="./licenses.txt"
                rel="noreferrer noopener"
                target="_blank"
              >
                Copyright Notice
              </a>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="c16"
      />
    </div>
  </div>
</body>
`;
